src/arrayMethodReduce.js

```diff
@@ -4,9 +4,18 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, ...initialValue) {
+    let accumulator = initialValue.length > 0
+      ? initialValue[0]
+      : this[0];
+    let startIndex = initialValue.length ? 0 : 1;
```

Comment:
The ternary operator usage for setting startIndex can lead to incorrect results if initialValue is undefined. You need to check if initialValue is not undefined before checking its length.

Reply:
@davinnchii  Don't forget to answer comments if you don't agree.